[DatePicker] Duplicate types exported by @mui/x-date-pickers/themeAugmentation

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Running `tsc` on my project results in these two errors:

```
../node_modules/@mui/lab/themeAugmentation/index.d.ts:5:1 - error TS2308: Module './overrides' has already exported a member named 'LabComponentNameToClassKey'. Consider explicitly re-exporting to resolve the ambiguity.

5 export * from '@mui/x-date-pickers/themeAugmentation';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../node_modules/@mui/lab/themeAugmentation/index.d.ts:5:1 - error TS2308: Module './props' has already exported a member named 'LabComponentsPropsList'. Consider explicitly re-exporting to resolve the ambiguity.

5 export * from '@mui/x-date-pickers/themeAugmentation';
```

I verified that this file is indeed exporting duplicate types with those names.

My package.json includes:

```json
        "@mui/lab": "^5.0.0-alpha.77",
        "@mui/material": "^5.6.1",
```

### Expected behavior 🤔

There should be no naming collisions and no TS errors

### Steps to reproduce 🕹

Steps:

1. `yarn add @mui/lab @mui/material`
2. `tsc`

### Context 🔦

It's a Next.js project. It's using Yarn workspaces and MUI is only contained in one of the workspaces.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: macOS 10.15.7
  Binaries:
    Node: 12.22.12 - ~/.nvm/versions/node/v12.22.12/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 6.14.16 - ~/.nvm/versions/node/v12.22.12/bin/npm
  Browsers:
    Chrome: 100.0.4896.75
    Edge: Not Found
    Firefox: 83.0
    Safari: 15.3
  npmPackages:
    @emotion/react:  11.9.0
    @emotion/styled:  11.8.1
    @mui/base:  5.0.0-alpha.76
    @mui/lab:  5.0.0-alpha.77
    @mui/material:  5.6.1
    @mui/private-theming:  5.6.1
    @mui/styled-engine:  5.6.1
    @mui/system:  5.6.1
    @mui/types:  7.1.3
    @mui/utils:  5.6.1
    @mui/x-date-pickers:  5.0.0-alpha.0
    @types/react: ^17.0.43 => 17.0.43
    react: ^17.0.2 => 17.0.2
    react-dom: ^17.0.2 => 17.0.2
    typescript: ^4.6.3 => 4.6.3
```
</details>

--- a/packages/x-date-pickers-pro/src/themeAugmentation/components.d.ts
+++ b/packages/x-date-pickers-pro/src/themeAugmentation/components.d.ts
@@ -1,6 +1,6 @@
 import { ComponentsProps, ComponentsOverrides, ComponentsVariants } from '@mui/material/styles';
 
-export interface PickerComponents {
+export interface PickersProComponents {
   MuiDateRangePickerDay?: {
     defaultProps?: ComponentsProps['MuiDateRangePickerDay'];
     styleOverrides?: ComponentsOverrides['MuiDateRangePickerDay'];
@@ -9,5 +9,5 @@ export interface PickerComponents {
 }
 
 declare module '@mui/material/styles' {
-  interface Components extends PickerComponents {}
+  interface Components extends PickersProComponents {}
 }

--- a/packages/x-date-pickers-pro/src/themeAugmentation/overrides.d.ts
+++ b/packages/x-date-pickers-pro/src/themeAugmentation/overrides.d.ts
@@ -1,12 +1,12 @@
 import { DateRangePickerDayClassKey } from '../DateRangePickerDay';
 
 // prettier-ignore
-export interface LabComponentNameToClassKey {
+export interface PickersProComponentNameToClassKey {
   MuiDateRangePickerDay: DateRangePickerDayClassKey;
 }
 
 declare module '@mui/material/styles' {
-  interface ComponentNameToClassKey extends LabComponentNameToClassKey {}
+  interface ComponentNameToClassKey extends PickersProComponentNameToClassKey {}
 }
 
 // disable automatic export

--- a/packages/x-date-pickers-pro/src/themeAugmentation/props.d.ts
+++ b/packages/x-date-pickers-pro/src/themeAugmentation/props.d.ts
@@ -1,11 +1,11 @@
 import { DateRangePickerDayProps } from '../DateRangePickerDay';
 
-export interface LabComponentsPropsList {
+export interface PickersProComponentsPropsList {
   MuiDateRangePickerDay: DateRangePickerDayProps<unknown>;
 }
 
 declare module '@mui/material/styles' {
-  interface ComponentsPropsList extends LabComponentsPropsList {}
+  interface ComponentsPropsList extends PickersProComponentsPropsList {}
 }
 
 // disable automatic export

--- a/packages/x-date-pickers/src/themeAugmentation/overrides.d.ts
+++ b/packages/x-date-pickers/src/themeAugmentation/overrides.d.ts
@@ -7,7 +7,7 @@ import { YearPickerClassKey } from '../YearPicker';
 import { PickerStaticWrapperClassKey } from '../internals/components/PickerStaticWrapper';
 
 // prettier-ignore
-export interface LabComponentNameToClassKey {
+export interface PickersComponentNameToClassKey {
   MuiCalendarPicker: CalendarPickerClassKey;
   MuiCalendarPickerSkeleton: CalendarPickerSkeletonClassKey;
   MuiClockPicker: ClockPickerClassKey;
@@ -20,7 +20,7 @@ export interface LabComponentNameToClassKey {
 }
 
 declare module '@mui/material/styles' {
-  interface ComponentNameToClassKey extends LabComponentNameToClassKey {}
+  interface ComponentNameToClassKey extends PickersComponentNameToClassKey {}
 }
 
 // disable automatic export

--- a/packages/x-date-pickers/src/themeAugmentation/props.d.ts
+++ b/packages/x-date-pickers/src/themeAugmentation/props.d.ts
@@ -17,7 +17,7 @@ import { TimePickerProps } from '../TimePicker';
 import { YearPickerProps } from '../YearPicker';
 import { PickerStaticWrapperProps } from '../internals/components/PickerStaticWrapper';
 
-export interface LabComponentsPropsList {
+export interface PickersComponentsPropsList {
   MuiCalendarPicker: CalendarPickerProps<unknown>;
   MuiCalendarPickerSkeleton: CalendarPickerSkeletonProps;
   MuiClockPicker: ClockPickerProps<unknown>;
@@ -39,7 +39,7 @@ export interface LabComponentsPropsList {
 }
 
 declare module '@mui/material/styles' {
-  interface ComponentsPropsList extends LabComponentsPropsList {}
+  interface ComponentsPropsList extends PickersComponentsPropsList {}
 }
 
 // disable automatic export

